retry.go

```diff
@@ -1,5 +1,40 @@
 package lo
 
+import (
+	"sync"
+	"time"
+)
+
+type Debounce struct {
+	after time.Duration
+	mu    *sync.Mutex
+	timer *time.Timer
+	done  bool
+}
+
+func (d *Debounce) Add(f func()) *Debounce {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+	if d.done {
+		return d
+	}
+	if d.timer != nil {
+		d.timer.Stop()
+	}
+	d.timer = time.AfterFunc(d.after, f)
```

Comment:
Also, maybe a nitpick, but what about the invocation time of the function itself? So say the function takes 50ms to run and the debounce is set to 100ms. Will the function be invoked after 150ms again? Should this be run in a goroutine so we can make sure it can be run again after 100ms?

Reply:
What if I have something like `.Add(f).Add(f)` and f takes 120ms to run? I couldn't invoke f again but I should be able to.

Reply:
> 
I think this debounce function will be executed once in set time(100ms), if the function is called again in set time(100ms), it will cancel the previous and reset. So no matter how much time the f is taken, it will be triggered **once** after specified time(100ms)
May be it is different from throttle..